sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py

```diff
@@ -2086,3 +2086,51 @@ def pipeline_func(job_in_number, job_in_path, component_func):
             component_func=component_func2)
         assert len(pipeline.jobs) == 2
         assert component_func2.name in pipeline.jobs
+
+    def test_pipeline_with_variable_inputs(self):
+        path = "./tests/test_configs/components/helloworld_component.yml"
+        component_func1 = load_component(path)
+        data = Data(name="test", version="1", type=AssetTypes.MLTABLE)
+
+        @dsl.pipeline
+        def pipeline_with_variable_args(*args, **kwargs):
+            node_args = component_func1(component_in_number=args[0], component_in_path=args[1])
+            node_kwargs = component_func1(component_in_number=kwargs["component_in_number1"],
+                                          component_in_path=kwargs["component_in_path1"])
+
+        @dsl.pipeline
+        def root_pipeline(component_in_number: int, component_in_path: Input, *args, **kwargs):
+            node = component_func1(component_in_number=component_in_number, component_in_path=component_in_path)
+            node_args = component_func1(component_in_number=args[0], component_in_path=args[1])
+            node_kwargs = component_func1(component_in_number=kwargs["component_in_number1"],
+                                          component_in_path=kwargs["component_in_path1"])
+            node_with_arg_kwarg = pipeline_with_variable_args(*args, **kwargs)
+
+        pipeline = root_pipeline(10, data, 11, data, component_in_number1=11, component_in_path1=data)
+
+        assert len(pipeline.inputs) == 6
+        arg_names = ['component_in_number', 'component_in_path', 'component_in_number1',
+                     'component_in_path1', 'args_0', 'args_1']
```

Comment:
Add some case that there is conflict parameter name
def pipepline(args_0, *args)

def pipepline(*args, **kwargs):
   pass
pipeline(list, arg_0 =xx)

Reply:
Add the test case that duplicate parameter name